src/dev/lupio/lupio_blk.hh

```diff
@@ -32,16 +32,16 @@
 #include "debug/LupioBLK.hh"
 #include "dev/dma_device.hh"
 #include "dev/io_device.hh"
+#include "dev/lupio/lupio_pic.hh"
 #include "dev/platform.hh"
-#include "dev/riscv/plic.hh"
 #include "dev/storage/disk_image.hh"
 #include "params/LupioBLK.hh"
 #include "sim/system.hh"
 
 namespace gem5
 {
 
-class Plic;
+class LupioPIC;
```

Comment:
Same question as before: do we actually need the blk and tty know about the PIC? I thought the interrupts were raised by calling the platform code directly, which then calls the PIC in a kind of generic way. Please correct me if I'm wrong.